CMakeCommon.cmake

```diff
@@ -99,10 +99,80 @@ function (LinkGSLibrariesToProject target acVersion devKitDir)
 
 endfunction ()
 
-function (GenerateAddOnProject target acVersion devKitDir addOnName addOnSourcesFolder addOnResourcesFolder addOnLanguage)
+function (parse_version inValue outList)
+    set (v1 0)
+    set (v2 0)
+    set (v3 0)
+    unset (CMAKE_MATCH_COUNT)
+    if (inValue MATCHES [[^([0-9]+)\.([0-9]+)\.([0-9]+)$]])
+    elseif (inValue MATCHES [[^([0-9]+)\.([0-9]+)$]])
+    elseif (inValue MATCHES [[^([0-9]+)$]])
+    endif ()
+    if (DEFINED CMAKE_MATCH_COUNT)
+        foreach (i RANGE 1 "${CMAKE_MATCH_COUNT}")
+            set ("v${i}" "${CMAKE_MATCH_${i}}")
+        endforeach ()
+        set ("${outList}" "${v1};${v2};${v3}" PARENT_SCOPE)
+    else ()
+        unset ("${outList}" PARENT_SCOPE)
+    endif ()
+endfunction ()
+
+function (generate_add_on_version_info)
+    parse_version ("${addOnVersion}" vers)
+    if (NOT DEFINED vers)
+        message (FATAL_ERROR "'${addOnVersion}' does not follow the '123' or '1.23' or '1.2.3' version format.")
+    endif ()
+    string (TIMESTAMP copyright "Copyright © ${addOnCompanyName}, ${addOnCopyrightYear}")
+
+    if (WIN32)
+        # FIXME(HVA): include GS build num in Windows release as well
+        list (APPEND vers "${acVersion}")
+        list (JOIN vers , versionComma)
+        list (JOIN vers . version)
+
+        set (out "${CMAKE_CURRENT_BINARY_DIR}/${target}-VersionInfo.rc")
+        configure_file ("${CMAKE_CURRENT_FUNCTION_LIST_DIR}/VersionInfo.rc.in" "${out}" @ONLY)
+        target_sources ("${target}" PRIVATE "${out}")
+    else ()
+        # BE on the safe side; load the info from an existing framework
+        file (READ "${devKitDir}/Frameworks/GSRoot.framework/Versions/A/Resources/Info.plist" plist_content NEWLINE_CONSUME)
+        string (REGEX MATCH "GSBuildNum[^0-9]+([0-9]+)" unused "${plist_content}")
+        set (gsBuildNum "${CMAKE_MATCH_1}")
+        string (REGEX MATCH "LSMinimumSystemVersion[^0-9]+([0-9.]+)" unused "${plist_content}")
+        set (lsMinimumSystemVersion "${CMAKE_MATCH_1}")
+
+        list (JOIN vers . shortVersion)
+
+        math (EXPR combined "${acVersion} * 10000 + ${gsBuildNum}")
+        list (APPEND vers "${combined}")
+        list (JOIN vers . longVersion)
+        if (NOT AC_ADDON_FOR_DISTRIBUTION)
+            string (APPEND longVersion -private)
+        endif ()
```

Comment:
I didn't have a better idea to mark a build as private on MacOS. This version is not the one that is displayed when a user brings up the context menu for the file and selects `Get Info`.

This can checked by Archicad later and maybe warn that there are private addons loaded. I'm just thinking of potential uses for this information. Right now there are no semantics attached to this.

Reply:
There is already a `GSPrivateBuild` key with value 1 in the Info.plist file of the add-ons we build during the normal jam-based process. Can you please try to use that?